File: src/main/scala/extracells/util/HandlerUniversalWirelessTerminal.scala
package extracells.util

import appeng.api.features.IWirelessTermHandler
import appeng.api.util.IConfigManager
import extracells.api.{IWirelessGasTermHandler, IWirelessFluidTermHandler}
import extracells.item.ItemWirelessTerminalUniversalScala
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack


object HandlerUniversalWirelessTerminal extends IWirelessTermHandler with IWirelessFluidTermHandler with IWirelessGasTermHandler{
  override def getConfigManager(is: ItemStack): IConfigManager = ItemWirelessTerminalUniversalScala.getConfigManager(is)

  override def canHandle(is: ItemStack): Boolean = ItemWirelessTerminalUniversalScala.canHandle(is)

  override def usePower(player: EntityPlayer, amount: Double, is: ItemStack): Boolean = ItemWirelessTerminalUniversalScala.usePower(player, amount, is)

  override def hasPower(player: EntityPlayer, amount: Double, is: ItemStack): Boolean = ItemWirelessTerminalUniversalScala.hasPower(player, amount, is)

  override def isItemNormalWirelessTermToo(is: ItemStack): Boolean = ItemWirelessTerminalUniversalScala.isItemNormalWirelessTermToo(is)

  override def setEncryptionKey(item: ItemStack, encKey: String, name: String): Unit = ItemWirelessTerminalUniversalScala.setEncryptionKey(item, encKey, name)

  override def getEncryptionKey(item: ItemStack): String = ItemWirelessTerminalUniversalScala.getEncryptionKey(item)
}

File: legacy/src/main/scala/extracells/util/HandlerUniversalWirelessTerminal.java
package extracells.util;

import appeng.api.features.IWirelessTermHandler;
import appeng.api.util.IConfigManager;
import extracells.api.IWirelessFluidTermHandler;
import extracells.item.ItemWirelessTerminalUniversal;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;

public class HandlerUniversalWirelessTerminal implements IWirelessFluidTermHandler, IWirelessTermHandler {

    public static ItemWirelessTerminalUniversal wireless = new ItemWirelessTerminalUniversal();
    
    @Override
    public boolean canHandle(ItemStack is) {
        return wireless.canHandle(is);
    }

    @Override
    public boolean usePower(EntityPlayer player, double amount, ItemStack is) {
        return wireless.usePower(player, amount, is);
    }

    @Override
    public boolean hasPower(EntityPlayer player, double amount, ItemStack is) {
        return wireless.hasPower(player, amount, is);
    }

    @Override
    public boolean isItemNormalWirelessTermToo(ItemStack is) {
        return wireless.isItemNormalWirelessTermToo(is);
    }

    @Override
    public IConfigManager getConfigManager(ItemStack is) {
        return wireless.getConfigManager(is);
    }

    @Override
    public String getEncryptionKey(ItemStack item) {
        return wireless.getEncryptionKey(item);
    }

    @Override
    public void setEncryptionKey(ItemStack item, String encKey, String name) {
        wireless.setEncryptionKey(item, encKey, name);
    }
}